validation-api/api/app.py

```diff
@@ -1,7 +1,51 @@
-from flask import Flask
+import json
+from http import HTTPStatus
+
+from api.utils import load_schema
+from flask import Flask, Response, request
+from jsonschema import validate
+from jsonschema.exceptions import SchemaError, ValidationError
 
 app = Flask(__name__)
 
+
 @app.route("/")
-def hello_world():
-    return "<p>Hello, World!</p>"
\ No newline at end of file
+def base():
+    """Root server route to ensure proper startup.
+    Returns:
+        str: Startup message.
+    """
+    return "<p>Config validation server running.</p>"
+
+
+@app.post("/validate")
+def validate_config():
+    """Server route to validate a user-supplied JSON configuration.
+    The route accepts a POST request with the configuration in the
+    request body. Then, it conducts logic to:
+    1. Load the schema (locally or from Azure storage)
+    2. Validate the configuration against the schema,
+    3. Return the configuration if it's valid, otherwise raise an HTTPError
+
+    Returns:
+        flask.Response: response object with valid JSON or error message.
+    """
+
+    # Load schema; locally for now. Return early if loading fails
+    try:
+        schema = load_schema(local=True)
+    except (json.decoder.JSONDecodeError, FileNotFoundError) as err:
+        return Response(response=str(err), status=HTTPStatus.BAD_REQUEST)

```

Comment:
Should this be a 400 error? There's an error with the server and nothing is wrong with or can be changed in the request, so maybe a 500?

Reply:
I could go either way but I would lean toward 400 here. Status codes are definitely not a science but in my experience generally 500 refers to a server failure (downtime, or something unexpected), whereas a 400 would indicate that the server is running fine but the request needs to be modified.